src/js/jsx/search/SearchBar.jsx

```diff
@@ -407,20 +420,24 @@ define(function (require, exports, module) {
          * Perform action based on ID
          *
          * @private
-         * @type {Datalist~onInputChange}
+         * @param {string} id of the selected option
          */
         _handleChange: function (id) {
+            var filters = this.state.filter,
+                filtersString,
+                idSplit,
+                type;
+
             if (id === null) {
+                type = "";
                 this.props.dismissDialog();
                 return;
+            } else {
```

Comment:
We prefer the following pattern:
```
if (badState) {
  cleanup();
  return;
}

// Normal case
...
```

Instead of:
```
if (badCase) {
  cleanup();
  return;
} else {
 // do stuff
}
```
We prefer the former because subjectively we find it easier to read by dismissing the exceptional cases first, and then proceeding with the main body at a lower indentation level.